### ISSUE TITLE
Things going where they're not supposed to

### ISSUE THREAD

**Issue by 61352151511** [RS485/LogisticsPipes#860]
So I've played with logistics pipes for a very long time. I know quite a bit about it and can't figure out why cobblestone and other "Go to the barrels or get trashed" materials are ending up in the diamond chests.

![2015-11-22_14 59 57](https://cloud.githubusercontent.com/assets/2399411/11325675/0462c6ee-912a-11e5-9c2f-eb5ec0acd647.png)
![2015-11-22_15 00 05](https://cloud.githubusercontent.com/assets/2399411/11325676/06984b50-912a-11e5-8b1a-56b950c0e479.png)

On the far left ender chest in the first picture is where all the materials enter the system. If it is an ore, then it goes into the second ender chest, if it is any material shown in the barrels in the second picture, it is supposed to go into the third ender chest connected with a "Logistics System Inventory Connector", all of the barrels have an itemsink module behind it, and the trash can has a mk5 pipe with item sink modules for all of the items in the barrels. For whatever reason though the diamond chests still get materials such as dirt, cobble, sand, etc. The diamond chests simply have a chassis mk2 with an unfiltered itemsink module, set to default route.

This situation didn't appear to happen before I switched the trash can from a basic pipe, to a mk5 chassis. At first I thought it had something to do with items being put into the trash can too fast for it to handle, so I created a little chest buffer, however the materials still end up in diamond chests.

If there's an explanation to why this happens and what I can do to get my system to work as intended, it would be appreciated. Never ran into this problem before and hope I can solve it because I don't really like my chests being full of cobble.

**Comment by davboecki** [comment]
By changing form a default pipe to a chassis pipe you changed from a dump pipe to an inteligent one when it comes to space calculation. When you have a basic pipe it doesn't care about the space the inventory has, but when there is a chassis pipe it checks to see if there is space in the inventory. This includes items that are already on the way to the chassis pipe.
You should probably use terminus modules on the trash as well as a quicksort module on your default route. That way if cobble ends up inside the default chest it will be sorted as soon as there "is space" for it.

**Comment by 61352151511** [comment]
Ok so, I put a Terminus module (Only cobblestone inside) into the Mk5 Chassis, the chest it's connected to is completely empty. I placed a QuickSort module into 3 of the diamond chests so far. As the quicksort module is looping over the slots I can see it going over the cobblestone and not moving it out of the diamond chests into the trash chest. However with the ModBased ItemSink module in the Mk5 the QuickSort sees that things from fossils & Archaeology get trashed, and sends them away. It just ignores the fact that cobble should be taken out of the chest for whatever reason.

**Comment by davboecki** [comment]
And you are sure that it's THAT cobblestone that is supposed to be moved to the terminus module. No damage value or cobble from another mod?
Have you checked if there is a firewall pipe in your system somewhere?

**Comment by 61352151511** [comment]
Yep, the only cobblestone in the system is regular old cobblestone. It works fine to send it to the barrel and it's just having trouble taking it out with the quicksort module to be sent to the terminus module. There's no firewall pipe, the entire system is what can be seen in the two pictures. I placed some upgrades in the barrel so that it's able to hold more and the quicksort module takes it out fine and sends it to the barrel, it just refuses to send it to the chest.

**Comment by mikeloeven** [comment]
I just use basic pipe on a DSU for cobble dirt and stone instead of barrels they are a bit expensive but you get 4 of them each time you make the recipie and they store nearly infinite items and have no annoying upgrades Like JABBA than you eliminate the overflow alltogether

have it pile up for a few weeks and once you can get into IC2 those dsu's can provide scrap for days

you can also try adding chassis with more slots and including an adv extractor on the diamond chests set to make sure any cobble that ends up there gets sent back for sorting

**Comment by 61352151511** [comment]
Well that's what the quicksort module is supposed to do, it's supposed to see that cobble has another place to be and send it back out, but it won't. I'd rather not use DSUs over the barrels because I like the look of them.

**Comment by loordgek** [comment]
put void upgrade on the barrel
